### PR TITLE
support variable-length string type for index

### PR DESCRIPTION
**Pull request by bright-starry-sky**

### CODE REVIEW

**Review by panda-sheep — APPROVED** [review]
LGTM

File: src/common/interface/meta.thrift [hunk 1]
```diff
@@ -574,11 +574,17 @@ struct HBReq {
     5: binary     git_info_sha
 }
 
+struct IndexFieldDef {
+    1: required binary       name,
+    // type_length is required if the field type is STRING.
+    2: optional i16          type_length,
```
Comment (panda-sheep) [inline src/common/interface/meta.thrift:580]: Excuse me, for the variable string,  `type_length`  is this length the maximum length of the field, similar to mysql's varchar(10)?
If the length of string is less than this length,  need to  pad 0 to the specified length?

**Review by panda-sheep — APPROVED** [review]

**Comment by bright-starry-sky** [comment]
This PR is related to the following two PRs, better to merge them together to avoid compilation errors. thanks.
https://github.com/vesoft-inc/nebula-graph/pull/316
https://github.com/vesoft-inc/nebula-storage/pull/156

File: src/common/interface/meta.thrift (hunk 1 above)
Comment (bright-starry-sky) [reply src/common/interface/meta.thrift:580]: > Excuse me, for the variable string, `type_length` is this length the maximum length of the field, similar to mysql's varchar(10)?
> If the length of string is less than this length, need to pad 0 to the specified length?

Yeah, as you say. 
When the field is string(10) of index:
if value length < 10 : need to pad '\0' to 10.
if value length > 10 : need to intercept a string of length 10.

Comment (panda-sheep) [reply src/common/interface/meta.thrift:580]: Excellent

**Comment by bright-starry-sky** [comment]
rebased

**Review by darionyaphet — APPROVED** [review]

**Review by panda-sheep — APPROVED** [review]

**Pull request merged**